packages/daemon/src/runner/job-runner.ts

```diff
@@ -117,55 +126,66 @@ export class JobRunner {
    * @param dataDir data dir
    * @param modelPath model path
    */
-  async runDataCollect(dataDir: string, modelPath: string): Promise<any> {
+  async runDataCollect(dataDir: string, modelPath: string, jobParam: IJobParam[]): Promise<any> {
     this.assertPlugin('dataCollect');
+    const param = this.getJobParam('dataCollect', jobParam);
+
     // ensure the model dir exists
     await fs.ensureDir(modelPath);
     // run dataCollect to download dataset.
     await this.runPlugin('dataCollect', this.getParams(this.opts.plugins.dataCollect.params, {
       dataDir
-    }));
+    }, ...param));
   }
 
   /**
    * run data access plugin
    * @param dataDir data dir
    */
-  async runDataAccess(dataDir: string): Promise<any> {
+  async runDataAccess(dataDir: string, jobParam: IJobParam[]): Promise<any> {
     this.assertPlugin('dataAccess');
+
+    const param = this.getJobParam('dataAccess', jobParam);
+
     return this.runPlugin('dataAccess', this.getParams(this.opts.plugins.dataAccess.params, {
       dataDir
-    }));
+    }, ...param));
   }
 
   /**
    * run dataset process plugin
    * @param dataset dataset from data collect plugin
    */
-  async runDatasetProcess(dataset: any): Promise<void> {
+  async runDatasetProcess(dataset: any, jobParam: IJobParam[]): Promise<void> {
     if (this.opts.plugins.datasetProcess) {
-      await this.runPlugin('datasetProcess', dataset, this.getParams(this.opts.plugins.datasetProcess.params));
```

Comment:
The job runner should always read the parameters from `JobEntity`, the pipeline parameters might be corresponding to the pipeline file, and should be unrelated to jobs.

Reply:
You are suggesting we should not explicitly use `param` from pipeline in the `job-runner`?